src/server/game/Entities/Object/Position.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Position.h"
+#include "ByteBuffer.h"
+#include "GridDefines.h"
+#include "Geometry.h"
+
+#include <G3D/g3dmath.h>
+#include <sstream>
+
+bool Position::operator==(Position const& a) const
+{
+    return (G3D::fuzzyEq(a.m_positionX, m_positionX) &&
+            G3D::fuzzyEq(a.m_positionY, m_positionY) &&
+            G3D::fuzzyEq(a.m_positionZ, m_positionZ) &&
+            G3D::fuzzyEq(a.m_orientation, m_orientation));
+}
+
+void Position::RelocatePolarOffset(float angle, float dist, float z /*= 0.0f*/)
+{
+    SetOrientation(GetOrientation() + angle);
+
+    m_positionX = GetPositionX() + dist * std::cos(GetOrientation());
+    m_positionY = GetPositionY() + dist * std::sin(GetOrientation());
+    m_positionZ = GetPositionZ() + z;
+}
+
+bool Position::HasInLine(Position const* pos, float width) const
+{
+    if (!HasInArc(float(M_PI), pos))
+        return false;
+
+    float angle = GetRelativeAngle(pos);
+    return std::fabs(std::sin(angle)) * GetExactDist2d(pos->GetPositionX(), pos->GetPositionY()) < width;
+}
+
+std::string Position::ToString() const
+{
+    std::stringstream sstr;
```

Comment:
For the future, it is better to use `Acore::StringFormatFmt`. It's 20 times faster